### PR TITLE
improvement: check static dir exists

### LINKED ISSUE #1127: Better error message when `static` dir is missing
Currently, Garden fails with `spawn git ENOENT` if it can't find the `static` directory. This can happen for users who install Garden manually and remove the `static` dir or otherwise don't co-locate it with the Garden binary.

This has caused confusion a few times. We should explicitly check for the `static` dir and throw a helpful error if it's not found.

### PR DESCRIPTION
**Pull request by dcharbonnier**
close #1127

### CODE REVIEW

File: garden-service/src/cli/cli.ts [hunk 1]
```diff
@@ -315,6 +316,9 @@ export class GardenCli {
           // tslint:disable-next-line: no-floating-promises
           checkForUpdates(garden.globalConfigStore, headerLog)
 
+          // tslint:disable-next-line: no-floating-promises
```
Comment (edvald) [inline garden-service/src/cli/cli.ts]: This comment should be unnecessary.

File: garden-service/src/cli/helpers.ts [hunk 2]
```diff
@@ -216,6 +217,15 @@ export function failOnInvalidOptions(argv, ctx) {
   }
 }
 
+export async function checkForStaticDir() {
+  if (!await pathExists(STATIC_DIR)) {
```
Comment (edvald) [inline garden-service/src/cli/helpers.ts]: This is a common gotcha (I've been burned many times myself), but you need to put parentheses around `await pathExists(STATIC_DIR)`. Or put the result in a variable and then check that, maybe easier to read.

File: garden-service/src/cli/helpers.ts [hunk 3]
```diff
@@ -216,6 +217,15 @@ export function failOnInvalidOptions(argv, ctx) {
...
+    throw new InternalError(
+      `The Garden static directory is missing, reinstall Garden.`,
```
Comment (edvald) [inline garden-service/src/cli/helpers.ts]: The phrasing could be a touch more helpful. Something like "Could not find the static data directory. Garden is packaged with a data directory called 'static', which should be located next to your garden binary. Please try reinstalling, and make sure the release archive is fully extracted to the target directory."

File: garden-service/src/cli/helpers.ts (hunk 2 above)
Comment (dcharbonnier) [reply garden-service/src/cli/helpers.ts]: the precedence of the await and the not are the same with an associativity to the left so this is correct but for clarity I change it

**Pull request merged**